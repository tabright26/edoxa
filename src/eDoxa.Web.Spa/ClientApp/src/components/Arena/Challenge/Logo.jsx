import React from "react";
import { Card } from "react-bootstrap";

import logo from "../../../assets/img/games/LeagueOfLegends.png";

const ArenaChallengeLogo = () => (
  <Card className="bg-dark mx-3 my-4">
    <Card.Img className="p-4" src={logo} />
  </Card>
);

export default ArenaChallengeLogo;
