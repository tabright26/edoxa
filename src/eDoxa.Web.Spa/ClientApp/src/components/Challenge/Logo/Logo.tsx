import React, { FunctionComponent } from "react";
import { Card, CardImg } from "reactstrap";
import logo from "assets/img/arena/games/leagueoflegends/small.png";

interface Props {
  readonly className?: string;
  readonly height?: number;
  readonly width?: number;
}

const Logo: FunctionComponent<Props> = ({ className, height = 200, width = 200 }) => (
  <Card className={className}>
    <CardImg className="p-4 m-auto" src={logo} style={{ height, width }} />
  </Card>
);

export default Logo;
