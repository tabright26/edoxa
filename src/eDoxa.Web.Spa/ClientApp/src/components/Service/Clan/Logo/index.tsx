import React, { Fragment, FunctionComponent } from "react";

import LogoForm from "./Form";

const imageStyle = {
  height: "100%",
  width: "100%"
};

const ClanLogo: FunctionComponent<any> = ({
  actions,
  logo,
  clanId,
  isOwner
}) => {
  return (
    <Fragment>
      <img
        style={imageStyle}
        src={logo ? logo : "https://via.placeholder.com/350x150"}
        alt="Team logo"
      ></img>
      {isOwner && (
        <LogoForm.Upload onSubmit={data => actions.updateLogo(clanId, data)} />
      )}
    </Fragment>
  );
};

export default ClanLogo;
