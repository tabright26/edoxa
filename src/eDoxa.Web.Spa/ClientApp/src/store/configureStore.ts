import { createStore, compose, applyMiddleware } from "redux";
import { loadUser } from "redux-oidc";
import { loadDoxaTags } from "store/doxaTags/actions";
import { middleware as thunkMiddleware } from "middlewares/thunkMiddleware";
import { middleware as routerMiddleware } from "middlewares/routerMiddleware";
import { middleware as axiosMiddleware } from "middlewares/axiosMiddleware";
import { middleware as signalrMiddleware } from "middlewares/signalrMiddleware";
import { middleware as loggerMiddleware } from "middlewares/loggerMiddleware";
import userManager from "utils/userManager";
import rootReducer from "./reducers";
import { RootState } from "./types";

// This enables the webpack development tools such as the Hot Module Replacement.
const composeEnhancers = (window["__REDUX_DEVTOOLS_EXTENSION_COMPOSE__"] as typeof compose) || compose;

export default (initialState: RootState) => {
  const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunkMiddleware, axiosMiddleware, signalrMiddleware, routerMiddleware, loggerMiddleware)));
  loadUser(store, userManager);
  const action: any = loadDoxaTags();
  store.dispatch(action);
  return store;
};
