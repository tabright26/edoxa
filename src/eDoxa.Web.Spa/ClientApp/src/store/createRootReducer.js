import { combineReducers } from "redux";

import { reducer as router } from "../middlewares/routerMiddleware";
import { reducer as form } from "redux-form";
import { reducer as modal } from "redux-modal";
import { reducer as oidc } from "redux-oidc";
import { reducer as stripe } from "../reducers/stripe/reducer";
import { reducer as arena } from "../reducers/arena/reducer";
import { reducer as user } from "../reducers/user/reducer";
import { reducer as doxaTags } from "../reducers/doxaTags/reducer";

const rootReducer = () =>
  combineReducers({
    router,
    form,
    modal,
    oidc,
    stripe,
    arena,
    user,
    doxaTags
  });

export default rootReducer;
