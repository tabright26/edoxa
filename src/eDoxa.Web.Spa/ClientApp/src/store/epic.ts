import { combineEpics, ofType } from "redux-observable";
import { NEVER } from "rxjs";
import { switchMap } from "rxjs/operators";
import {
  CREATE_USER_TRANSACTION_SUCCESS,
  CREATE_USER_TRANSACTION_FAIL,
  REDEEM_PROMOTION_FAIL,
  REDEEM_PROMOTION_SUCCESS
} from "store/actions/cashier/types";
import {
  CREATE_USER_ADDRESS_SUCCESS,
  DELETE_USER_ADDRESS_SUCCESS,
  UPDATE_USER_ADDRESS_SUCCESS,
  CREATE_USER_ADDRESS_FAIL,
  DELETE_USER_ADDRESS_FAIL,
  UPDATE_USER_ADDRESS_FAIL,
  LOGIN_USER_ACCOUNT_SUCCESS,
  LOGIN_USER_ACCOUNT_FAIL,
  REGISTER_USER_ACCOUNT_SUCCESS,
  REGISTER_USER_ACCOUNT_FAIL
} from "store/actions/identity/types";
import {
  CHANGE_USER_DOXATAG_SUCCESS,
  CHANGE_USER_DOXATAG_FAIL
} from "store/actions/identity/types";
import {
  CREATE_USER_PROFILE_SUCCESS,
  UPDATE_USER_PROFILE_SUCCESS,
  CREATE_USER_PROFILE_FAIL,
  UPDATE_USER_PROFILE_FAIL
} from "store/actions/identity/types";
import {
  FORGOT_USER_PASSWORD_SUCCESS,
  RESET_USER_PASSWORD_SUCCESS,
  FORGOT_USER_PASSWORD_FAIL,
  RESET_USER_PASSWORD_FAIL
} from "store/actions/identity/types";
import {
  UPDATE_USER_PHONE_SUCCESS,
  UPDATE_USER_PHONE_FAIL
} from "store/actions/identity/types";
import {
  VALIDATE_GAME_AUTHENTICATION_FAIL,
  VALIDATE_GAME_AUTHENTICATION_SUCCESS,
  GENERATE_GAME_AUTHENTICATION_SUCCESS,
  GENERATE_GAME_AUTHENTICATION_FAIL,
  UNLINK_GAME_CREDENTIAL_SUCCESS,
  UNLINK_GAME_CREDENTIAL_FAIL
} from "./actions/game/types";

const formSuccessEpic = (action$: any): any =>
  action$.pipe(
    ofType(
      CREATE_USER_ADDRESS_SUCCESS,
      DELETE_USER_ADDRESS_SUCCESS,
      UPDATE_USER_ADDRESS_SUCCESS,
      CHANGE_USER_DOXATAG_SUCCESS,
      CREATE_USER_PROFILE_SUCCESS,
      UPDATE_USER_PROFILE_SUCCESS,
      FORGOT_USER_PASSWORD_SUCCESS,
      RESET_USER_PASSWORD_SUCCESS,
      UPDATE_USER_PHONE_SUCCESS,
      CREATE_USER_TRANSACTION_SUCCESS,
      VALIDATE_GAME_AUTHENTICATION_SUCCESS,
      GENERATE_GAME_AUTHENTICATION_SUCCESS,
      UNLINK_GAME_CREDENTIAL_SUCCESS,
      REDEEM_PROMOTION_SUCCESS,
      LOGIN_USER_ACCOUNT_SUCCESS,
      REGISTER_USER_ACCOUNT_SUCCESS
    ),
    switchMap((action: any): any => {
      const { resolve } = action.meta.previousAction.meta;
      if (resolve) {
        resolve(action.payload);
      }
      return NEVER;
    })
  );

const formFailEpic = (action$: any) =>
  action$.pipe(
    ofType(
      CREATE_USER_ADDRESS_FAIL,
      DELETE_USER_ADDRESS_FAIL,
      UPDATE_USER_ADDRESS_FAIL,
      CHANGE_USER_DOXATAG_FAIL,
      CREATE_USER_PROFILE_FAIL,
      UPDATE_USER_PROFILE_FAIL,
      FORGOT_USER_PASSWORD_FAIL,
      RESET_USER_PASSWORD_FAIL,
      UPDATE_USER_PHONE_FAIL,
      CREATE_USER_TRANSACTION_FAIL,
      VALIDATE_GAME_AUTHENTICATION_FAIL,
      GENERATE_GAME_AUTHENTICATION_FAIL,
      UNLINK_GAME_CREDENTIAL_FAIL,
      REDEEM_PROMOTION_FAIL,
      LOGIN_USER_ACCOUNT_FAIL,
      REGISTER_USER_ACCOUNT_FAIL
    ),
    switchMap((action: any) => {
      const { reject } = action.meta.previousAction.meta;
      if (reject) {
        reject(action.error);
      }
      return NEVER;
    })
  );

export const epic = combineEpics(formSuccessEpic, formFailEpic);
