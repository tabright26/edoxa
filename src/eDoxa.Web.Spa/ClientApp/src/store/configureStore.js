import { createStore, compose, applyMiddleware } from "redux";
import { loadUser } from "redux-oidc";
import { loadDoxaTags } from "actions/identity/actionCreators";
import { middleware as thunkMiddleware } from "middlewares/thunkMiddleware";
import { middleware as routerMiddleware } from "middlewares/routerMiddleware";
import { middleware as axiosMiddleware } from "middlewares/axiosMiddleware";
import { middleware as signalrMiddleware } from "middlewares/signalrMiddleware";
import { middleware as loggerMiddleware } from "middlewares/loggerMiddleware";
import userManager from "utils/userManager";
import createRootReducer from "./createRootReducer";

// This enables the webpack development tools such as the Hot Module Replacement.
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default initialState => {
  const store = createStore(createRootReducer(), initialState, composeEnhancers(applyMiddleware(thunkMiddleware, axiosMiddleware, signalrMiddleware, routerMiddleware, loggerMiddleware)));
  loadUser(store, userManager);
  store.dispatch(loadDoxaTags());
  return store;
};
