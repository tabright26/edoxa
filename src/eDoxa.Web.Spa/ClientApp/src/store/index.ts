import { createStore, compose, applyMiddleware } from "redux";

import { RootState } from "store/types";

import { reducer as rootReducer } from "store/reducer";

import { middleware as thunkMiddleware } from "utils/thunk/middleware";
import { middleware as routerMiddleware } from "utils/router/middleware";
import { middleware as axiosMiddleware } from "utils/axios/middleware";
import { middleware as signalrMiddleware } from "utils/signalr/middleware";
import { middleware as loggerMiddleware } from "utils/logger/middleware";

import userManager from "utils/oidc/userManager";

import { loadUser } from "redux-oidc";

// This enables the webpack development tools such as the Hot Module Replacement.
const composeEnhancers =
  (window["__REDUX_DEVTOOLS_EXTENSION_COMPOSE__"] as typeof compose) || compose;

export const configureStore = (initialState: RootState) => {
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(
        thunkMiddleware,
        axiosMiddleware,
        signalrMiddleware,
        routerMiddleware,
        loggerMiddleware
      )
    )
  );
  loadUser(store, userManager);
  return store;
};
