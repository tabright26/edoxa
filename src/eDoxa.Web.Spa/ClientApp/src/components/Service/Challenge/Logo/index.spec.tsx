import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import Logo from ".";
import store from "store";

it("renders without crashing", () => {
  // Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <Logo />
      </Provider>
    )
    .toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});
