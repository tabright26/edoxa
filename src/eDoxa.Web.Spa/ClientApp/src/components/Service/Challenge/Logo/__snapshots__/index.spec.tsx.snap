// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="my-0 h-100 bg-gray-900 null card"
>
  <img
    className="p-4 m-auto card-img"
    src="small.png"
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
</div>
`;
